src/rust/engine/src/externs/interface.rs

```diff
@@ -1389,6 +1395,12 @@ fn session_record_test_observation(
   })
 }
 
+fn session_isolated_shallow_clone(py: Python, session_ptr: PySession) -> CPyResult<PySession> {
+  with_session(py, session_ptr, |session| {
+    PySession::create_instance(py, session.isolated_shallow_clone())
+  })
+}
```

Comment:
Generally, I am curious if we'll be able to move free functions like this onto methods defined on `PySession`? Totally okay to not do in this PR - it's a followup project I want to explore to see if it further simplifies our FFI.

Reply:
Yes, we probably can. The `with_$x` context managers add a bunch of noise, and are much less central than calling `allow_threads`, which we should try to do in nearly all cases.

Reply:
Opened https://github.com/pantsbuild/pants/issues/11722